platform_channels/lib/src/method_channel_demo.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:platform_channels/src/counter_method_channel.dart';
+
+/// The widget demonstrates how to use [MethodChannel] to invoke platform methods.
+/// It has two [FloatingActionButton]s to increment and decrement the value of
+/// [count], and a [Text] widget to display the value it.
+class MethodChannelDemo extends StatefulWidget {
+  @override
+  _MethodChannelDemoState createState() => _MethodChannelDemoState();
+}
+
+class _MethodChannelDemoState extends State<MethodChannelDemo> {
+  int count = 0;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('MethodChannel Demo'),
+      ),
+      floatingActionButton: Builder(
+        builder: (context) {
+          return Column(
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              FloatingActionButton(
+                heroTag: 'addition',
+                child: Icon(Icons.add),
+
+                // Whenever users presses the FloatingActionButton, it invokes
+                // Counter.increment method to increment the value of count.
+                onPressed: () {
+                  Counter.increment(counterValue: count).then((value) {
+                    setState(() => count = value);
+                  }).catchError((dynamic error) {
+                    showErrorMessage(
+                      context,
+                      error.message as String,
+                    );
+                  });
+                },
```

Comment:
I believe I had a comment in this section earlier, but just in case: typically I would expect to see an async method here. Generally we recommend using async/await over the Futures API whenever possible. 

The logic can all stay the same; it's just a syntactical change:

```dart
onPressed: () async {
  try {
    final value = await Counter.increment(counterValue: count);
    setState(() => count = value);
  } catch (error) {
    showErrorMessage(
      context,
      error.message as String,
    );
  }
},
```

Reply:
I missed it in the previous commit after doing the changes.